pyartifactory/objects/build.py

```diff
@@ -0,0 +1,206 @@
+
+        try:
+            response = self._get(
+                f"api/{self._uri}/{build_name}/{build_number}{_query_string}",
+            )
+            self._response_checker(response)
+            logger.debug("Build Info successfully retrieved")
+            return BuildInfo(**response.json())
+        except requests.exceptions.HTTPError as error:
+            http_response: Union[Response, None] = error.response
+            if isinstance(http_response, Response) and http_response.status_code == 404:
+                raise BuildNotFoundError(f"Build {build_number} were not found on {build_name}")
+            raise ArtifactoryError from error
+        except ArtifactoryBuildError as error:
+            if error.status == 404:
+                raise BuildNotFoundError(error.message)
+            raise ArtifactoryError from error
+
+    def create_build(self, create_build_request: BuildCreateRequest):
+        try:
+            _existing_build_info = self.get_build_info(create_build_request.name, create_build_request.number)
+        except BuildNotFoundError:
+            _resp = self._put(f"api/{self._uri}", json=create_build_request.model_dump())
+            if _resp.status_code != 204:
+                logger.error("Build %s in %s not created", create_build_request.number, create_build_request.name)
+                raise ArtifactoryError(
+                    f"Build {create_build_request.number} in {create_build_request.name} not created",
+                )
+            logging.debug("Build %s in %s successfully created", create_build_request.number, create_build_request.name)
+        else:
+            logger.error("Build %s in %s already exists", create_build_request.number, create_build_request.name)
+            raise ArtifactoryError(f"Build {create_build_request.number} in {create_build_request.name} already exists")
+
+    def promote_build(
+        self,
+        build_name: str,
+        build_number: str,
+        promotion_request: BuildPromotionRequest,
+    ) -> BuildPromotionResult:
+        """
+        :param build_name: Build name to be promoted
+        :param build_number: Build number to be promoted
+        :param promotion_request: Model object containing parameters for promotion
+        :return: BuildPromotionResponse containing server response
+        """
+        try:
+            response = self._get(
+                f"api/{self._uri}/{build_name}/{build_number}",
+            )
+            self._response_checker(response)
+        except requests.exceptions.HTTPError as error:
+            http_response: Union[Response, None] = error.response
+            if isinstance(http_response, Response) and http_response.status_code == 404:
+                raise BuildNotFoundError(f"Build {build_number} were not found on {build_name}")
+            raise ArtifactoryError from error
+        except ArtifactoryBuildError as error:
+            if error.status == 404:
```

Comment:
Seems to be a duplicate; 404 is already raised earlier

Reply:
same reason as below